source/data-modeling/field-types.txt

```diff
@@ -0,0 +1,558 @@
+   :end-before: # end-datetime-string
+
+.. note::
+
+   {+odm+} parses string values into ``DateTime`` by using the ``Time.parse``
+   method, which considers values without time zones to be in local time.
+
+Timestamps
+~~~~~~~~~~
+
+You can include timestamp fields by including the ``Mongoid::Timestamps``
+module when you create your class. Timestamp fields include the following
+fields:
+
+- ``created_at``: Stores the time the document was created.
+- ``updated_at``: Stores the time the document was last updated.
+
+The following example creates a ``Person`` class with timestamp fields:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-timestamps
+   :end-before: # end-timestamps
+
+You can also choose to include only the ``created_at`` or ``updated_at`` fields
+by including only the ``Created`` or ``Updated`` modules. The following example
+creates a ``Person`` class with only the ``created_at`` field, and a ``Post``
+class with only the ``updated_at`` field:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-timestamps-specific
+   :end-before: # end-timestamps-specific
+
+You can shorten the timestamp field names to ``c_at`` and ``u_at`` by setting
+the ``::Short`` option when including the module:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-timestamps-short
+   :end-before: # end-timestamps-short
+
+You can disable creating the timestamp field for specific operations by calling
+the ``timeless`` method on the method call. The following example disables the
+timestamps for the ``save`` operation:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-timestamps-disable
+   :end-before: # end-timestamps-disable
+
+Regexp
+~~~~~~
+
+You can store regular expressions in a field by using the ``Regexp`` type. 
+
+While MongoDB uses `Perl Compatible Regular Expressions (PCRE) <https://www.pcre.org/>`__,
+{+odm+} uses Ruby's `Onigmo <https://github.com/k-takata/Onigmo>`__. PCRE and
+Onigmo provide generally similar functionality, but there are several syntax
+differences. For example, Onigmo uses ``\A`` and ``\z`` to match the beginning and
+end of a string, while PCRE uses ``^`` and ``$``.
+
+When you declare a field as a ``Regexp``, {+odm+} converts Ruby regular
+expressions to BSON regular expressions when storing the result into your
+database. The database returns the field as a ``Bson::Regexp::Raw`` instance.
+You can use the ``compile`` method on ``BSON::Regexp::Raw`` instances to convert
+the data back to a Ruby regular expression.
+
+The following example creates a ``Token`` class and specifies the ``pattern``
+field as a ``Regexp``:
+
+.. literalinclude:: /includes/data-modeling/field-types.rb
+   :language: ruby
+   :start-after: # start-regexp
+   :end-before: # end-regexp
+
```

Comment:
S: can link to regex section of specify a query guide

Reply:
Added as a TODO since I don't have that version of the specify page on my branch (and trying to pull now is causing too many issues)